### PR TITLE
Pairix recipe

### PR DESCRIPTION
**Pull request by nvictus**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

Adding a new recipe for [pairix](https://github.com/4dn-dcic/pairix) by @SooLee and @carlvitzthum. Please review, @bioconda/core.  Thanks.

### CODE REVIEW

File: recipes/pairix/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+requirements:
+  build:
+    - gcc
+    - zlib
+    - python
+    - setuptools
+
+  run:
+    - python
+    - htslib
```
Comment (bgruening) [inline recipes/pairix/meta.yaml]: Mh, is zlib really needed as build dependency if you don't have it in the run dependency?

File: recipes/pairix/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+  fn: {{ version }}.tar.gz
+  url: https://github.com/4dn-dcic/pairix/archive/{{ version }}.tar.gz
+  md5: 05ac0be28a85e4d6bdca744ddec8824e
+
+build:
+  number: 0
+  
+requirements:
+  build:
+    - gcc
```
Comment (bgruening) [inline recipes/pairix/meta.yaml]: Can you use `gcc  # [linux]` and `llvm  # [osx]`

File: recipes/pairix/meta.yaml (hunk 1 above)
Comment (bgruening) [inline recipes/pairix/meta.yaml]: Why is htslib needed and not in the build deps?

File: recipes/pairix/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  run:
+    - python
+    - htslib
+
+test:
+  imports:
+    - pypairix
+  commands:
+    - which pairix
```
Comment (bgruening) [inline recipes/pairix/meta.yaml]: Is something like `parix --help` possible?

**Comment by bgruening** [comment]
Thanks @nvictus?

File: recipes/pairix/meta.yaml (hunk 2 above)
Comment (nvictus) [inline recipes/pairix/meta.yaml]: Will do!

File: recipes/pairix/meta.yaml (hunk 1 above)
Comment (nvictus) [inline recipes/pairix/meta.yaml]: Pairix is meant to work with bgzip, which is provided by htslib. Pairix builds a copy of the bgzip executable, but I figured it is better to use the official htslib version and overwrite it.

File: recipes/pairix/meta.yaml (hunk 3 above)
Comment (nvictus) [inline recipes/pairix/meta.yaml]: I couldn't find a benign option that returns an exit code of 0. @SooLee, does one exist or would it be possible to add a `--help` option that exits with 0?

Originally I tried to pull in sample files and run a test using:
```
  source_files:
    - samples/test_4dn.pairs.gz
    - samples/test_4dn.pairs.gz.px2
  commands:
    - pairix -l samples/test_4dn.pairs.gz
```
but never got it to work. The build system couldn't find the source files.

**Comment by daler** [comment]
@nvictus you can test for other exit codes, e.g., https://github.com/bioconda/bioconda-recipes/blob/master/recipes/asciigenome/meta.yaml#L19

**Comment by nvictus** [comment]
Thanks, @daler! They ended up adding a `--help` option that exits with 0 in the end, but great to know.

**Comment by bgruening** [comment]
Cool, thanks!

**Pull request merged**